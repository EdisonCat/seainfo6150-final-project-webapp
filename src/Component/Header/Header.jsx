import React, { useState, useContext } from "react";
import styles from "./Header.module.css";
import { Link, useHistory } from "react-router-dom";
import logo from "../../images/logo.png";
import MarvelFandomContext from "../../Context/MarvelFandomContext";
export default function Header(props) {
	const [keyword, setKeyword] = useState("");
	const history = useHistory();
	const context = useContext(MarvelFandomContext);
	const handleHome = (e) => {
		e.preventDefault();
		history.push({ pathname: "/" });
	};
	const handleSearch = () => {
		history.push({ pathname: `/characters/${keyword}` });
	};
	return (
		<div className={styles.container}>
			<div className={styles.upper}>
				{props.loggedIn ? (
					<label className={styles.greeting}>
						Hello, {props.username}
					</label>
				) : (
					<label className={styles.greeting}>
						The best Marvel Fandom in the world
					</label>
				)}
				<img
					className={styles.logo}
					src={logo}
					alt="logo"
					onClick={(e) => handleHome(e)}
				/>
				<div className={styles.search}>
					<form>
						<input
							type="text"
							placeholder="Type the id of the character"
							value={keyword}
							onChange={(e) => setKeyword(e.target.value)}
						/>
						<button onClick={(e) => handleSearch(e)}>Search</button>
					</form>
				</div>
			</div>
			<div className={styles.menu}>
				<img
					className={styles.menubutton}
					src="https://img.icons8.com/cotton/64/000000/menu.png"
					alt="menu"
					width="32"
					style={props.sidebar ? { transform: "rotate(90deg)" } : {}}
					onClick={() => props.setSidebar(!props.sidebar)}
				/>
				<Link to="/" className={styles.link}>
					Home
				</Link>
				<Link to="/characters" className={styles.link}>
					Characters
				</Link>
				<Link to="/comics" className={styles.link}>
					Comics
				</Link>
				{props.loggedIn ? (
					<Link
						onClick={() => context.logout()}
						to="/"
						className={styles.link}
					>
						Log Out
					</Link>
				) : (
					<>
						<Link to="/signup" className={styles.link}>
							Sign Up
						</Link>
						<Link to="/login" className={styles.link}>
							Log In
						</Link>
					</>
				)}
			</div>
		</div>
	);
}