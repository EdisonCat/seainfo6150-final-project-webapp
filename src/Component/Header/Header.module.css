@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300&display=swap');
.container {
	margin: 0;
	padding: 0;
	width: 100%;
	display: flex;
	background-color: #333;
	min-height: 80px;
	align-items: center;
	justify-content: center;
	flex-direction: column;
	color: white;
	font-family: 'Montserrat', sans-serif;
}
.upper {
	display: flex;
	border-bottom: 1px grey solid;
	width: 100%;
	justify-content: space-around;
	align-items: center;
	height: fit-content;
}
.greeting,
.logo,
.search {
	display: flex;
	justify-content: center;
	width: 500px;
}
.logo {
    max-width: 100%;
    width: 120px;
	max-height: 50px;
    height: auto;
	cursor: pointer;
	height: fit-content;
}
.search input {
	outline: none;
	width: 200px;
	transition: 0.5s;
}
.search input:hover {
	width: 220px;
}
.menu {
	display: flex;
	align-items: center;
	height: fit-content;
}
.link {
	text-decoration: underline;
	margin: 5px 20px 5px 20px;
	outline: none;
	color: white;
	transition: 0.5s;
	text-decoration-color: #333;
}
.link:hover {
	text-decoration-color: red;
}
.menubutton {
	display: none;
	cursor: pointer;
	transition: 0.5s;
}
@media (max-width: 800px) {
	.greeting,
	.search {
		display: none;
	}
}
@media (max-width: 600px) {
	.menubutton {
		display: block;
	}
	.link {
		display: none;
	}
}