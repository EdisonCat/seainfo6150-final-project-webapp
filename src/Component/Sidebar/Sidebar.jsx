import React, {useContext} from "react";
import { Link } from "react-router-dom";
import styles from "./Sidebar.module.css";
import MarvelFandomContext from "../../Context/MarvelFandomContext";
export default function Sidebar(props) {
    const context = useContext(MarvelFandomContext);
    const handleLogout = () => {
        context.logout();
        props.setSidebar(false);
    }
	return (
		<div
			className={styles.container}
			style={props.sidebar ? { left: "0px", top: "113px" } : {}}
		>
			<Link
				className={styles.link}
				to="/"
				onClick={() => props.setSidebar(false)}
			>
				Home
			</Link>
			{props.loggedIn ? (
				<Link className={styles.link} to="/" onClick={() => handleLogout()}>
					Log Out
				</Link>
			) : (
				<>
					<Link
						className={styles.link}
						to="/signup"
						onClick={() => props.setSidebar(false)}
					>
						Sign Up
					</Link>
					<Link
						className={styles.link}
						to="/login"
						onClick={() => props.setSidebar(false)}
					>
						Log In
					</Link>
				</>
			)}
			<Link
				className={styles.link}
				to="/characters"
				onClick={() => props.setSidebar(false)}
			>
				Characters
			</Link>
			<Link
				className={styles.link}
				to="/comics"
				onClick={() => props.setSidebar(false)}
			>
				Comics
			</Link>
		</div>
	);
}
