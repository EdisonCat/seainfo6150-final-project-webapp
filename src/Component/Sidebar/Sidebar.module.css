@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@100&family=Zen+Dots&display=swap');
.container {
    background-color: rgba(22, 22, 22, 0.8);
    transition: 0.5s;
    z-index: 1;
    display: flex;
    flex-direction: column;
    align-items: center;
    color: white;
    position: absolute;
    left: -100%;
    top: 113px;
    width: 100%;
    height: 100vh;
	font-family: 'Montserrat', sans-serif;
}
.link {
    text-decoration: underline;
    text-decoration-color: transparent;
    outline: none;
    color: white;
    width: 100%;
    text-align: center;
    margin: 20px;
    transition: 0.5s;
}
.link:hover {
    text-decoration-color: red;
}