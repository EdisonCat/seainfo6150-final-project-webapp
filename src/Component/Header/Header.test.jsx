import React from "react";
import Header from "./Header";
import Sidebar from "../Sidebar/Sidebar";
it("rendered Header correctly", () => {
	// eslint-disable-next-line no-undef
	const { container } = render(<Header Sidebar={Sidebar} />);
	expect(container).toMatchSnapshot();
});