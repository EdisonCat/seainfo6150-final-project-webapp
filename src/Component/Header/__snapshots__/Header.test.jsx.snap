// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Header correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="upper"
    >
      <label
        class="greeting"
      >
        The best Marvel Fandom in the world
      </label>
      <img
        alt="logo"
        class="logo"
        src="logo.png"
      />
      <div
        class="search"
      >
        <form>
          <input
            placeholder="Type the id of the character"
            type="text"
            value=""
          />
          <button>
            Search
          </button>
        </form>
      </div>
    </div>
    <div
      class="menu"
    >
      <img
        alt="menu"
        class="menubutton"
        src="https://img.icons8.com/cotton/64/000000/menu.png"
        width="32"
      />
      <a
        class="link"
        href="/"
      >
        Home
      </a>
      <a
        class="link"
        href="/characters"
      >
        Characters
      </a>
      <a
        class="link"
        href="/comics"
      >
        Comics
      </a>
      <a
        class="link"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="link"
        href="/login"
      >
        Log In
      </a>
    </div>
  </div>
</div>
`;
